runtime/etc/Makefile.mli-shared

```diff
@@ -19,44 +19,25 @@
 # Makefile.mli-shared
 #
 
-include $(CHPL_MAKE_HOME)/runtime/etc/Makefile.include
+include $(CHPL_MAKE_HOME)/runtime/etc/Makefile.mli-common
 
-ifdef CHPL_ZMQ_HOME
-CHPL_MLI_INCS = -I$(CHPL_ZMQ_HOME)/include
-CHPL_ZMQ_LIB_HOME = $(CHPL_ZMQ_HOME)/lib
-CHPL_ZMQ_LIB_SEARCH_PATH = -L$(CHPL_ZMQ_LIB_HOME)
-endif
-CHPL_LN_LIB_DIR = $(CHPL_MAKE_RUNTIME_LIB)/$(CHPL_MAKE_LAUNCHER_SUBDIR)
-UNAME = $(shell uname)
-
-.PHONY: buildUserObj buildServer buildClient runTagsCommand
-
-all: buildUserObj buildServer buildClient runTagsCommand FORCE
-
-buildUserObj:
-#
-# Compile loose C source files in CHPLUSEROBJ.
-#
-	$(foreach srcFile, $(CHPLUSEROBJ), \
-		$(CC) $(CHPL_MAKE_BASE_CFLAGS) $(GEN_CFLAGS) $(COMP_GEN_CFLAGS) \
-			-c -o $(srcFile) $(CHPL_RT_INC_DIR) $(srcFile).c \
-	;)
+.PHONY: all
+all: checkRtLibDir buildServer buildClientAsHost runTagsCommand FORCE
 
-buildServer: $(TMPSERVERNAME)
+.PHONY: buildClientAsHost
+buildClientAsHost: $(TMPBINNAME)
 
+.PHONY: buildClient
 buildClient: $(TMPBINNAME)
 
-runTagsCommand:
-	$(TAGS_COMMAND)
-
-$(TMPBINNAME): $(CHPL_CL_OBJS) FORCE
 #
 # Invoke the launcher script, but don't actually link up the launcher binary.
-# Do this by setting CHPL_MAKE_IS_MLI=1. We also need to pass on the
-# $(GEN_CFLAGS) variable, because that contains the -fPIC flag, among other
-# things. If we don't do this, then the launcher object file that is built
-# will have text relocations.
+# Do this by setting CHPL_MAKE_IS_MLI=1 when invoking the sub-make. We also
+# need to pass on the $(GEN_CFLAGS) variable, because that contains the -fPIC
+# flag, among other things. If we don't do this, then the launcher object
+# file that is built will have text relocations.
```

Comment:
This ("file ...will have text relocations") is overly informative.  It would be better to just say that we're producing a shared object here, so the launcher object file needs to be PIC.  Then we won't have to assume that readers know what text relocations are and why they're a problem in this context.